New bitmap functions: `bitmapHasAny` and `bitmapHasAll`

svladykin opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
Added bitmap related functions `bitmapHasAny` and `bitmapHasAll` analogous to `hasAny` and `hasAll` functions for arrays.

svladykin committed:
Support for bitmapHasAny and bitmapHasAll functions.

svladykin committed:
minor tests fix

andyyzh commented:
From the clickhouse-server.txt, Crash when this use case is executed.

> 2019.05.16 01:09:04.003917 [ 52 ] {38de2a3c-cbd9-49b7-a055-7486559955ea} <Debug> executeQuery: (from [::1]:51410) select bitmapHasAny(bitmapBuild([     0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,     100,200,500]),bitmapBuild([ 99, 100, 101])); 
2019.05.16 01:09:04.004296 [ 1570 ] {} <Error> BaseDaemon: ########################################
2019.05.16 01:09:04.004337 [ 1570 ] {} <Error> BaseDaemon: (version 19.8.1.485) (from thread 52) Received signal Segmentation fault (11).
2019.05.16 01:09:04.004367 [ 1570 ] {} <Error> BaseDaemon: Address: NULL pointer.
2019.05.16 01:09:04.004377 [ 1570 ] {} <Error> BaseDaemon: Access: read.
2019.05.16 01:09:04.004386 [ 1570 ] {} <Error> BaseDaemon: Address not mapped to object.
2019.05.16 01:09:04.100914 [ 1570 ] {} <Error> BaseDaemon: 0. clickhouse-server(roaring_bitmap_intersect+0x99) [0x78c1dd9]
2019.05.16 01:09:04.100977 [ 1570 ] {} <Error> BaseDaemon: 1. clickhouse-server(DB::FunctionBitmapCardinality<DB::BitmapHasAnyImpl, DB::NameBitmapHasAny, unsigned char>::executeImpl(DB::Block&, std::vector<unsigned long, std::allocator<unsigned long> > const&, unsigned long, unsigned long)+0x66b) [0x3beea6b]

alexey-milovidov commented:
@svladykin Please check the failing tests.

alexey-milovidov reviewed (changes requested):
.

svladykin commented:
Thanks, will look!

alexey-milovidov reviewed (changes requested):
.

svladykin committed:
Merge branch 'master' of https://github.com/yandex/ClickHouse into bitmap1

svladykin committed:
Fixed types in tests.

svladykin committed:
Code style fixed.

svladykin commented:
@alexey-milovidov @andyyzh 
I've fixed the array element type related issues in tests, fixed the code style.
I can not reproduce the crash now:
```
$ clickhouse-dev client -n --testmode < 00829_bitmap_function.sql > 00829_bitmap_function.result
$ diff 00829_bitmap_function.result 00829_bitmap_function.reference 
$
```

yingfeng commented:
Corruption happened if arguments come from scalar subqueries, seen from [issue 5479](https://github.com/yandex/ClickHouse/issues/5479)
